instant-xml-macros/src/lib.rs

```diff
@@ -2,9 +2,39 @@ extern crate proc_macro;
 
 use proc_macro::TokenStream;
 use quote::quote;
+use std::collections::HashMap;
 use syn::{parse_macro_input, DeriveInput, Lit, Meta, NestedMeta};
 
-fn retrieve_default_namespace(input: &DeriveInput) -> Option<String> {
+type DefaultNamespace = String;
+type OtherNamespaces = HashMap<String, String>;
+type Namespaces = Option<(Option<DefaultNamespace>, Option<OtherNamespaces>)>;
+
+fn retrieve_namespaces(input: &DeriveInput) -> Namespaces {
+    let mut default_namespace = DefaultNamespace::default();
+    let mut other_namesapces = OtherNamespaces::default();
+
+    if let NestedMeta::Meta(Meta::List(list)) = get_meta_items(&input.attrs).first()? {
+        if list.path.get_ident()? == "namespace" {
+            for item in &list.nested {
+                match item {
+                    NestedMeta::Lit(Lit::Str(v)) => {
+                        default_namespace = v.value();
```

Comment:
Some error handling is needed here for the cases like: 
`#[xml(namespace("URI", "BAZ", foo = "BAR"))]`
As it will make the BAZ default namespace.